tests/internal/test_native.py

```diff
@@ -0,0 +1,59 @@
+from ddtrace.internal.native import get_configuration_from_disk
+
+
+def test_get_configuration_from_disk(tmp_path):
+    # First test -- config matches & should be returned
+    config_1 = tmp_path / "config_1.yaml"
+    config_1.write_text(
+        """
+rules:
+  - selectors:
+    - origin: language
+      matches:
+        - python
+      operator: equals
+    configuration:
+      DD_SERVICE: my-service
+""",
+        encoding="utf-8",
+    )
+
+    config = get_configuration_from_disk(file_override=str(config_1))
+    assert config == {"DD_SERVICE": "my-service"}
+
+    # Second test -- config does not match & should not be returned
+    config_2 = tmp_path / "config_2.yaml"
+    config_2.write_text(
+        """
+rules:
+  - selectors:
+    - origin: language
+      matches:
+        - nodejs
+      operator: equals
+    configuration:
+      DD_SERVICE: my-service
+""",
+        encoding="utf-8",
+    )
+
+    config = get_configuration_from_disk(file_override=str(config_2))
+    assert config == {}
+
+
+def is_imported_before(module_a, module_b):
+    import sys
+
+    # Check if both modules are in sys.modules
+    if module_a in sys.modules and module_b in sys.modules:
+        # Get the position of the modules in sys.modules (which is an OrderedDict in Python 3.7+)
+        modules = list(sys.modules.keys())
+        return modules.index(module_a) < modules.index(module_b)
+    return False  # If one or both modules are not imported, return False
+
+
+def test_native_before_settings():
```

Comment:
we should probably have a test that applying a configuration via file gets applied to the application.

do we expect to have system-tests for this stuff as well? (if not, can we?)

Reply:
> Besides, yes we want to add system-tests to test this behaviour on customer-like systems

Can we add those _before_ we merge this? It should be fully tested before we merge, since as soon as it is merged it can be released.

Reply:
> It should be fully tested before we merge, since as soon as it is merged it can be released.

Would you be ok with merging this under an experimental flag that disables calling to this code by default, so we are sure no customer uses it? It's hard to bootstrap system-tests if we don't have any merged language implementation to run the actual tests against.

We need to make sure that this code doesn't crash stuff obviously but this is something that can be tested in integration tests in ddtrace-py rather than in system-tests.

Reply:
You should be able to write tests against a dev version of this library pretty easily: https://github.com/DataDog/system-tests/blob/main/docs/execute/binaries.md#python-library

Merging tests in system-tests can be tricky since we need to ensure they are expected to fail on older versions.


I'd prefer the approach of writing at least the first pass of tests against this branch, but otherwise, a private/hidden feature flag would be ok.

Reply:
I've written a first system test and made it pass locally against this PR: https://github.com/DataDog/system-tests/pull/3982. Happy to get your review @brettlangdon. Let me know if there is anything to change on this PR to run this test.